### PR TITLE
492 progress bar

### PR DESCRIPTION
**Pull request by SkyRiser568**
Progress bar to show level completion.

📝 Checklist before requesting a review:

- [x]  I have run this branch locally
- [x]  I have performed a self-review of my code
- [x]  I have confirmed critical features still function

### CODE REVIEW

**Comment by SkyRiser568** [comment]
By the way, since progress bar and % ui are the two designs we're comparing on thursday. Does that mean we should hold off the merge till after we decide which route we want to take? Otherwise we might have to make two seperate scenes to test each one, unless im just confused

**Comment by CrispyKorn** [comment]
> By the way, since progress bar and % ui are the two designs we're comparing on thursday. Does that mean we should hold off the merge till after we decide which route we want to take? Otherwise we might have to make two seperate scenes to test each one, unless im just confused

Yeah I reckon we should wait till after class to merge them. Although It would suck to lose work so we might be able to merge them both and just disable the one we don't want. That way it can be reused later if we want.

File: MonstieWash/Assets/Scripts/UI/ProgressBarUI.cs
Comment (CrispyKorn) [inline MonstieWash/Assets/Scripts/UI/ProgressBarUI.cs:41]: literally the only thing is to use the `String.Equals()` method when working with strings since `==` can be unreliable.

```suggestion
        foreach (var task in m_taskData.Where(t => t.gameObject.scene.Equals(m_gameSceneManager.CurrentScene)))
```

**Review by CrispyKorn — APPROVED** [review]
Nice, everything looks good to me. Like I said though we'll hold off on merging for now, but I'll put this review here to get ahead of things.

File: MonstieWash/Assets/Scripts/UI/ProgressBarUI.cs
Comment (SkyRiser568) [reply MonstieWash/Assets/Scripts/UI/ProgressBarUI.cs:41]: With this, because im comparing Scene objects and not strings, is it still preferable to use String.Equals(). And also is there a benefit to comparing scenes using names/strings vs scene objects? Just trying to pick your brain for some knowledge.

Comment (CrispyKorn) [reply MonstieWash/Assets/Scripts/UI/ProgressBarUI.cs:41]: Oh lol I'm just blind then whoopsies. Dw about that then.

I had a look into the unity code for scenes, and turns out each scene has a unique identifier for each instance that it uses to do equality checks, and the identifier is just an int. This means that I think comparing scenes the way you did above is actually faster, so disregard my suggestion. I'll go back and look at changing the other scene comparisons in the codebase to do it this way at some point as well.

**Pull request merged**
